primitives/src/pcs/structs.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2022 Espresso Systems (espressosys.com)
+// This file is part of the Jellyfish library.
+
+// You should have received a copy of the MIT License
+// along with the Jellyfish library. If not, see <https://mit-license.org/>.
+
+use ark_ec::PairingEngine;
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize, Read, SerializationError, Write};
+
+#[derive(Derivative, CanonicalSerialize, CanonicalDeserialize)]
+#[derivative(
+    Default(bound = ""),
+    Hash(bound = ""),
+    Clone(bound = ""),
+    Copy(bound = ""),
+    Debug(bound = ""),
+    PartialEq(bound = ""),
+    Eq(bound = "")
+)]
+/// A commitment is an Affine point.
+pub struct Commitment<E: PairingEngine>(
+    /// the actual commitment is an affine point.
+    pub E::G1Affine,
```

Comment:
feel like I asked this question before, can @zhenfeizhang reminds me why we need this wrapper struct again?